source/faas-automated-testing/faas-automated-testing.rst

```diff
@@ -286,63 +286,75 @@ Supported Evergreen variants that have the AWS SAM CLI installed:
 - rhel82-arm64
 
 
-This is an example function in the Evergreen config that accomplishes this, using
-subprocess.exec to execute a script that calls the drivers-evergreen-tools
-function inside of it:
+This is an example task group in the Evergreen config that accomplishes this, using
+subprocess.exec to execute scripts that call the drivers-evergreen-tools
+functions inside of it for setup, teardown, and execution:
 
 .. code:: yaml
 
-  run deployed aws lambda tests:
-  - command: ec2.assume_role
-    params:
-      role_arn: ${LAMBDA_AWS_ROLE_ARN}
-  - command: subprocess.exec
-    params:
-      working_dir: src
-      binary: bash
-      args:
-        - .evergreen/run-deployed-lambda-aws-tests.sh
-      env:
-        TEST_LAMBDA_DIRECTORY: ${PROJECT_DIRECTORY}/test/lambda
-        DRIVERS_TOOLS: ${DRIVERS_TOOLS}
-        DRIVERS_ATLAS_PUBLIC_API_KEY: ${DRIVERS_ATLAS_PUBLIC_API_KEY}
-        DRIVERS_ATLAS_PRIVATE_API_KEY: ${DRIVERS_ATLAS_PRIVATE_API_KEY}
-        DRIVERS_ATLAS_LAMBDA_USER: ${DRIVERS_ATLAS_LAMBDA_USER}
-        DRIVERS_ATLAS_LAMBDA_PASSWORD: ${DRIVERS_ATLAS_LAMBDA_PASSWORD}
-        DRIVERS_ATLAS_GROUP_ID: ${DRIVERS_ATLAS_GROUP_ID}
-        LAMBDA_STACK_NAME: dbx-node-lambda
-        AWS_REGION: us-east-1
-        AWS_ACCESS_KEY_ID: ${AWS_ACCESS_KEY_ID}
-        AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY}
-        AWS_SESSION_TOKEN: ${AWS_SESSION_TOKEN}
+  tasks:
+    - name: "test-aws-lambda-deployed"
+      commands:
+        - func: "install dependencies"
+        - command: ec2.assume_role
+          params:
+            role_arn: ${LAMBDA_AWS_ROLE_ARN}
+            duration_seconds: 3600
+        - command: subprocess.exec
+          params:
+            working_dir: src
+            binary: bash
+            add_expansions_to_env: true
+            args:
+              - ${DRIVERS_TOOLS}/.evergreen/aws_lambda/run-deployed-lambda-aws-tests.sh
+            env:
+              TEST_LAMBDA_DIRECTORY: ${PROJECT_DIRECTORY}/test/lambda
+              AWS_REGION: us-east-1
+  task_groups:
+    - name: test_aws_lambda_task_group
+      setup_group:
+        - func: fetch source
+        - command: subprocess.exec
+          params:
+            working_dir: src
+            binary: bash
+            add_expansions_to_env: true
+            args:
+              - ${DRIVERS_TOOLS}/.evergreen/aws_lambda/setup-atlas-cluster.sh
+        - command: expansions.update
+          params:
+            file: src/atlas-expansion.yml
```

Comment:
What creates `atlas-expansion.yml`? I don't see it referenced anywhere in this RST file and "atlas-expansion" doesn't appear anywhere in the current master branches of drivers-evergreen-tools or the specs repo.

Per [expansions.update](https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Project-Commands#expansionsupdate), this will error if the file doesn't exist.

Reply:
Ah, found it in [mongodb-labs/drivers-evergreen-tools#325](https://github.com/mongodb-labs/drivers-evergreen-tools/pull/325/files#diff-bbc90d2c444efad446a3b080f0b6116258a7dc45366140be98ca8b891934c646R121-R122). The file is created by `setup-atlas-cluster.sh`.

Reply:
It's done in the accompanying tools PR here: https://github.com/mongodb-labs/drivers-evergreen-tools/pull/325/files#diff-bbc90d2c444efad446a3b080f0b6116258a7dc45366140be98ca8b891934c646R122